### PR TITLE
Print additional runtime details in Python utilities

### PR DESCRIPTION
**Pull request by tyxchen**
This PR resolves issue #2284.

In addition to adding printing of the runtime details to the console, I've also added printing of the details to the HTML reports generated by runtestsuite.py and the log generated by appleseed.benchmark. I've also slightly refactored appleseed.benchmark to use argparse for parsing arguments, and added version numbers to all of the utility scripts.

### CODE REVIEW

File: scripts/oslextractmeta.py [hunk 1]
```diff
@@ -337,16 +345,16 @@ def cli():
...
-        The JSON dictionary consists of a 'header' and a 'shader' part.
-        jsondict['shader'] will return a dictionary with all shaders. The
-        user interface of the shader is stored as a sub-dictionary, the
-        metadata can be retrieved using the 'ui' key on the elements, e.g.:
-
-        for x in jsondict['shaders'].values():
-            print x['ui']
-        ''')
+oslextractmetadata stores the user interface and metadata of a
+compiled OSL (openshadinglanguage) shader(s) into a JSON file.
```
Comment (oktomus) [inline scripts/oslextractmeta.py:349]: You will end up with `of acompiled OSL` here

Comment (tyxchen) [reply scripts/oslextractmeta.py:349]: Wouldn't the newline be preserved in a triple-quoted string?

Comment (oktomus) [reply scripts/oslextractmeta.py:349]: Good point. I honestly don't know ^^

Comment (tyxchen) [reply scripts/oslextractmeta.py:349]: Just confirmed, the newline is printed in the help text.

**Comment by tyxchen** [comment]
@oktomus Is there anything more that needs to be fixed in this PR?

File: scripts/appleseed.benchmark.py [hunk 2]
```diff
@@ -28,12 +28,16 @@
...
 
+from __future__ import print_function
+import argparse
 import datetime
 import os
 import re
 import subprocess
 import sys
 
+from utils import print_runtime_details  # local module
```
Comment (dictoon) [inline scripts/appleseed.benchmark.py:39]: Nice!

**Pull request merged**